calc

```diff
@@ -1,39 +1,17 @@
-
-mode=str(input('Input Operator'))
-op1=float(input('Input Operator 1'))
-op2=float(input('Input Operator 2'))
-
-
-
 def calculate(op1,op2,mode):
-    if mode=='add':
-        y=(op1+op2)
-    elif mode=='substract':
-        y=(op1-op2)
-    elif mode=='multiply':
-        y=(op1*op2)
-    elif mode=='divide':
-        y=(op1/op2)
-        
-    return(y)
+    switcher = {
+        'add': op1 + op2,
+        'substract': op1 - op2,
+        'multiply': op1 * op2,
+        'divide': op1 / op2
+    }
+
+    return(switcher.get(mode, "Invalid mode"))
```

Comment:
switch-case statement ist hier "schöner" als if else Abfrage.

Reply:
kommt das jetzt. 
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-10-eebf9b4347d4> in <module>
     18 
     19 # For division by 0
---> 20 my_result = calculate(2,0,"divide")
     21 assert_equal(my_result,None)
     22 
<ipython-input-9-b2e9d9d8bd0d> in calculate(op1, op2, mode)
     40         'subtract': op1 - op2,
     41         'multiply': op1 * op2,
---> 42         'divide': op1 / op2
     43     }
     44 
ZeroDivisionError: division by zero

Reply:
jetzt kommt das hier 

```suggestion
    switcher = {
        'add': op1 + op2,
        'substract': op1 - op2,
        'multiply': op1 * op2,
        'divide': op1 / op2
    }

    return(switcher.get(mode, "Invalid mode"))
```

Reply:
jetzt kommt das hier:
```suggestion
 ---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-10-eebf9b4347d4> in <module>
     18 
     19 # For division by 0
---> 20 my_result = calculate(2,0,"divide")
     21 assert_equal(my_result,None)
     22 
<ipython-input-9-b2e9d9d8bd0d> in calculate(op1, op2, mode)
     40         'subtract': op1 - op2,
     41         'multiply': op1 * op2,
---> 42         'divide': op1 / op2
     43     }
     44 
ZeroDivisionError: division by zero
```

Reply:
Also durch 0 teilen ist auch in python nicht möglich 😄

Reply:
hi again ich kann es nicht rein kopieren, ist gesperrt. bisher geht alles, ich muss jetzt nur noch irgendwie versuchen dem programm zu sagen, dass er nicht durch null teilen darf. ist dämlich echt haha